x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/mapper/SemanticTextInferenceResultFieldMapper.java

```diff
@@ -0,0 +1,241 @@
+import org.elasticsearch.index.mapper.SourceLoader;
+import org.elasticsearch.index.mapper.SourceValueFetcher;
+import org.elasticsearch.index.mapper.TextFieldMapper;
+import org.elasticsearch.index.mapper.TextSearchInfo;
+import org.elasticsearch.index.mapper.ValueFetcher;
+import org.elasticsearch.index.mapper.vectors.SparseVectorFieldMapper;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+
+// TODO: fill in blanks in javadoc
+/**
+ * A mapper for the {@code _semantic_text_inference} field.
+ * <br>
+ * <br>
+ * This mapper works in tandem with {@link SemanticTextFieldMapper semantic_text} fields to index inference results.
+ * The inference results for {@code semantic_text} fields are written to {@code _source} by ______ like so:
+ * <br>
+ * <br>
+ * <pre>
+ * {
+ *     "_source": {
+ *         "my_semantic_text_field": "these are not the droids you're looking for",
+ *         "_semantic_text_inference": {
+ *             "my_semantic_text_field": {
+ *                 [
+ *                     {
+ *                         "sparse_embedding": {
+ *                             "lucas": 0.05212344,
+ *                             "ty": 0.041213956,
+ *                             "dragon": 0.50991,
+ *                             "type": 0.23241979,
+ *                             "dr": 1.9312073,
+ *                             "##o": 0.2797593
+ *                         },
+ *                         "text": "these are not the droids you're looking for"
+ *                     }
+ *                 ]
+ *             }
+ *         }
+ *     }
+ * }
+ * </pre>
+ *
+ * This mapper parses the contents of the {@code _semantic_text_inference} field and indexes it as if the mapping were configured like so:
+ * <br>
+ * <br>
+ * <pre>
+ * {
+ *     "mappings": {
+ *         "properties": {
+ *             "my_semantic_text_field": {
+ *                 "type": "nested",
+ *                 "properties": {
+ *                     "sparse_embedding": {
+ *                         "type": "sparse_vector"
+ *                     },
+ *                     "text": {
+ *                         "type": "text",
+ *                         "index": false
+ *                     }
+ *                 }
+ *             }
+ *         }
+ *     }
+ * }
+ * </pre>
+ */
+public class SemanticTextInferenceResultFieldMapper extends MetadataFieldMapper {
+    public static final String CONTENT_TYPE = "_semantic_text_inference";
+    public static final String NAME = "_semantic_text_inference";
+    public static final String SPARSE_VECTOR_SUBFIELD_NAME = "sparse_embedding";
+    public static final String TEXT_SUBFIELD_NAME = "text";
+    public static final TypeParser PARSER = new FixedTypeParser(c -> new SemanticTextInferenceResultFieldMapper());
+
+    // TODO: Need to query this as a nested field type?
```

Comment:
We will need to query as a nested field, but that doesn't relate to FieldType, right? 🤔

Reply:
Correct, this was more of a note to myself to verify this point